### DISCUSSION TITLE
Autorun Docker

### DISCUSSION THREAD

**Discussion by seppelp** [aahnik/tgcf#483] [Q&A]
Hello,
can't figure out how to run tgcf on the start of the docker-container. Everything is working great except after a reboot of the server I need to manually start tgcf. I want it to run once the docker-container starts. Any suggestions? Thanks!

**Comment by MobCode100** [comment]
As of now, we can't do anything until the owner of this project gets active again. 
But if you're familiar with command line, I can help you for that.

Get the name of your container using `docker ps`, or if you're using docker desktop, then it's easier. Navigate to **Terminal** tab and just run the below commands but skip the first command.

First, run `docker exec -it <NAME_OF_CONTAINER> bash`

Next, run the below commands:

```
apt install nano
nano /etc/systemd/system/tgcf.service
```

Next copy and paste these lines inside the nano application:
```
[Unit]
Description=tgcf Telegram forwarder

[Service]
User=root
WorkingDirectory=/app/
ExecStart=/venv/bin/tgcf live
Restart=always

[Install]
WantedBy=multi-user.target
```
Change `live` with the mode you want tgcf to run. Save the file using `Ctrl + s` and exit `Ctrl + x`. 

Let's continue with:
```
systemctl daemon-reload
systemctl enable tgcf
systemctl start tgcf
```

Now you can check whether tgcf is running using `systemctl status tgcf`.
Should the container restarts, tgcf will also starts.

I hope I'm able to help, and  if that's the case, be sure to mark this reply as the answer thank you :smile:.

**Reply by seppelp** [reply]
Perfect, so I only need a service in the container. Thanks a lot for the great description of the service!
